sdk/keyvault/azure-security-keyvault-keys/test/telemetry_header_test.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#if defined(_MSC_VER)
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include "gtest/gtest.h"
+
+#include "mocked_transport_adapter_test.hpp"
+
+#include <azure/core/internal/strings.hpp>
+#include <azure/keyvault/key_vault.hpp>
+
+#include <string>
+
+using namespace Azure::Security::KeyVault::Keys::Test;
+
+TEST_F(MockedTransportAdapterTest, keyvaultTelemetryId)
+{
+  m_clientOptions.TelemetryPolicyOptions.ApplicationId = "ourApplicationId";
+  m_client = std::make_unique<
+      Azure::Security::KeyVault::Keys::Test::KeyClientWithNoAuthenticationPolicy>(
+      "url", m_clientOptions);
+
+  // The fake response from the mocked transport adapter is good for parsing a Key back
+  auto response = m_client->GetKey("name");
+
+  // The response is an echo of the sent headers. Let's find the telemetry id
+  auto foundHeader = false;
+  for (auto& header : response.GetRawResponse().GetHeaders())
+  {
+    if (Azure::Core::Internal::Strings::LocaleInvariantCaseInsensitiveEqual(
+            header.first, "User-Agent"))
+    {
+      foundHeader = true;
+      EXPECT_PRED2(
+          [](std::string const& received, std::string const& sent) {
+            auto telemetryInfoWithNoOSAndDate = received.substr(0, sent.size());
+            return Azure::Core::Internal::Strings::LocaleInvariantCaseInsensitiveEqual(
+                telemetryInfoWithNoOSAndDate, sent);
+          },
+          header.second,
+          "ourApplicationId azsdk-cpp-KeyVault/7.2");
```

Comment:
Hardcoding the `7.2` into the version is brittle.  This test will break each time we update the server api version.  I would expect m_client has a read only property for the KeyAPI version.  I'd use that to compare if the User-Agent is of expected formated.

Reply:
Actually, I am only interested in finding the applicationId in the start, so, I would remove looking at the next part of the header. We can have another test for checking that part.